### ISSUE TITLE
Getting error in GetProfileDataAsync 

### ISSUE THREAD

**Issue by wiseguytamo** [IdentityServer/IdentityServer4#3708]
Hello.

I am having an issue validating user login via IdentityServer4.
My front-end app is a Razor Pages app.
I have added a user with two role claims:
new Claim("role", "AppAdmin"), 
new Claim("role", "SysAdmin")

If the user is assigned to one role I am able to log in successfully. However, when assigned to
more than one roles I get the following exception:

![Exception](https://user-images.githubusercontent.com/11686469/66368344-f43b3380-e965-11e9-9a23-924fd4fd6c11.PNG)

---ProfileService class which implements IProfileService Interface ---

![ProfileService](https://user-images.githubusercontent.com/11686469/66368374-0ae18a80-e966-11e9-98aa-475cafd79f1d.PNG)

What am I missing or doing wrong ?

**Comment by leastprivilege** [comment]
Nothing is wrong in your IdentityServer code.

Microsoft's OIDC handler tries to do some claims mapping. Check the docs for their ClaimAction feature

**Comment by wiseguytamo** [comment]
Hello @leastprivilege .

Thank you very much for your suggestion, it helped me 100% and I was able to solve the issue I was experiencing.

The issue was concurring because of how I configured my ClaimActions in the Razor Pages client app.
A user can have more than one role claims but I was using " options.ClaimActions.MapUniqueJsonKey("role", "role") " in the OIDC configuration which caused the error.

Below is a screenshot that contains the commented code that was throwing the error. It is highlighted with " <-- CAUSED ERROR " on line 131.

 The fix  is highlighted with " <-- FIXED ERROR " on line 132.

![Error_Fix](https://user-images.githubusercontent.com/11686469/66446417-74ba6c80-ea18-11e9-808d-89bcc69b9bc5.PNG)


Thanks again @leastprivilege .

Regards,
